tests/IntegrationTest/Definitions/WildcardDefinitionsTest.php

```diff
@@ -38,6 +38,21 @@ public function test_wildcards(ContainerBuilder $builder)
         self::assertEntryIsNotCompiled($container, 'DI\Test\IntegrationTest\*\Interface*');
         self::assertEntryIsNotCompiled($container, Interface1::class);
     }
+    /**
+     * @dataProvider provideContainer
+     */
+    public function test_wildcard_with_static_name(ContainerBuilder $builder)
+    {
+        $builder->addDefinitions([
+            'DI\Test\IntegrationTest\*\Interface*' => \DI\create(Implementation1::class),
+        ]);
+        $container = $builder->build();
+
+        $object = $container->get(Interface1::class);
+        $this->assertInstanceOf(Implementation1::class, $object);
+
+        self::assertEntryIsNotCompiled($container, Interface1::class);
+    }
```

Comment:
I've added a simpler version of your test: this test passes on the current master. I think this is better to have a simple test (easy to grasp) which covers the base of the feature + your test that is more advanced and covers the bug you were facing.